pydax/_dataset.py

```diff
@@ -0,0 +1,215 @@
+#
+# Copyright 2020 IBM Corp. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"Download and load a dataset"
+
+
+from enum import IntFlag
+import hashlib
+import os
+import pathlib
+import tarfile
+from typing import Any, Dict, Iterable, Optional, Tuple, Union
+
+from packaging.version import parse as version_parser
+import requests
+
+from . import _typing, get_config
+from ._schema import load_schemata, SchemaDict
```

Comment:
```suggestion
from ._schema import load_schemata
from .schema import SchemaDict
```

So is it better here to import from where `SchemaDict` was defined or from where it was publicly exposed?

Reply:
Referencing this from before: https://github.com/CODAIT/pydax/pull/4/files?file-filters%5B%5D=.py#r525542962

Reply:
you're right, we should use the public version

Reply:
it looks uglier like that though because `load_schemata` can't be imported from `pydax` without causing a circular import error 😭